test/WidoCollateralSwap.t.sol

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.7;
+
+import "forge-std/Test.sol";
+import "forge-std/StdUtils.sol";
+import "@openzeppelin/contracts/utils/math/SafeMath.sol";
+import "../contracts/WidoCollateralSwap.sol";
+import "./mocks/MockSwap.sol";
+import "../contracts/interfaces/IComet.sol";
+import "./interfaces/ICometTest.sol";
+import "./ForkTest.sol";
+
+contract WidoCollateralSwapTest is ForkTest {
+    using SafeMath for uint256;
+    WidoCollateralSwap widoCollateralSwap;
+    MockSwap mockSwap;
+
+    IERC3156FlashLender constant flashLoanProvider = IERC3156FlashLender(0x4EAF187ad4cE325bF6C84070b51c2f7224A51321);
+    ICometTest constant cometUsdc = ICometTest(0xc3d688B66703497DAA19211EEdff47f25384cdc3);
+
+    WidoCollateralSwap.Collateral existingCollateral = WidoCollateralSwap.Collateral(WBTC, 0.06e8);
+    WidoCollateralSwap.Collateral finalCollateral = WidoCollateralSwap.Collateral(WETH, 1e18);
+
+    event SupplyCollateral(address indexed from, address indexed dst, address indexed asset, uint amount);
+    event WithdrawCollateral(address indexed src, address indexed to, address indexed asset, uint amount);
+
+    function setUp() public {
+        widoCollateralSwap = new WidoCollateralSwap(
+            flashLoanProvider
+        );
+        mockSwap = new MockSwap(
+            ERC20(WETH),
+            ERC20(WBTC)
+        );
+    }
+
+    function test_itWorks() public {
```

Comment:
Please add tests for the unhappy paths.

- What if allowSignature is incorrect.
- What if revokeSignature is incorrect.
- What if there is no positive slippage.
- What if there is negative slippage.